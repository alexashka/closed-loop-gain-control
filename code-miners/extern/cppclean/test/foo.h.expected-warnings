Processing foo.h
Unable to find dir//bar.h
foo.h:14: 'Unused' not used
foo.h:61: 'ProtoArray' not used
foo.h:113: not-used.h does not need to be #included
foo.h:150: 'AR' not used
