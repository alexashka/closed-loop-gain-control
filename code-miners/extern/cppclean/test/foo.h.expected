Processing foo.h
Include(1, 15, <map>)
Include(16, 33, <string>)
Include(35, 51, "bar.h")
Include(52, 75, "xss_module.h")
Typedef(86, 89, Mapper, Type(86, 94, std::map<[Type(95, 106, std::string), Type(108, 111, int)]>))
Typedef(136, 137, AnonStruct, [Struct(136, 137, None, None, [VariableDeclaration(138, 141, Type(138, 141, int) zz)])])
Typedef(185, 186, NamedPtr, [Union(185, 186, NamedUnion, {[Token('bool', 187, 191), Token('xx', 192, 194), Token(';', 194, 195)]})])
Class(216, 219, Baz, None, None)
Class(227, 233, Unused, None, None)
Class(241, 246, Local, None, None)
Class(254, 264, DataMember, None, None)
Class(272, 275, Ptr, None, None)
Class(284, 288, Data, None, None)
VariableDeclaration(290, 296, Type(290, 296, extern Data*) d)
Class(335, 337, Jj, None, None in somespace)
Class(367, 370, Foo, [Type(380, 383, Bar)], [Function(406, 409, None Foo([]), 0x08, None in FooNS), Function(415, 418, None Foo([Parameter(419, 450, Type(419, 427, register const unsigned long&) a), Parameter(452, 463, Type(452, 455, Baz*) output)]), 0x08, [] in FooNS), Function(479, 482, None Foo([]), 0x16, None in FooNS), Function(513, 518, Type(513, 533, const somespace::Jj&) Get([]), 0x03, None in FooNS), Function(549, 554, Type(549, 569, const somespace::Jj&) Get2([]), 0x05, None in FooNS), Function(590, 593, Type(590, 593, int) Get3([]), 0x01, [Token('return', 609, 615), Token('0', 616, 617), Token(';', 617, 618)] in FooNS), Function(645, 646, Type(645, 647, T&) foo([Parameter(652, 658, Type(652, 653, T*) bar)]), 0x00, None in FooNS), Function(674, 678, Type(674, 678, void) Method([Parameter(686, 706, Type(686, 691, const Mapper&) mapper), Parameter(708, 724, Type(708, 717, somespace::Jj*) j)]), 0x00, None in FooNS), Function(729, 735, Type(729, 736, Mapper&) Blah([]), 0x00, None in FooNS), VariableDeclaration(747, 753, Type(747, 753, Mapper) data_), VariableDeclaration(763, 773, Type(763, 773, DataMember*) data), VariableDeclaration(783, 791, Type(783, 791, auto_ptr<[Type(792, 795, Ptr*)]>) ptr)] in FooNS)
Function(807, 811, Type(807, 811, void) Function([Parameter(821, 835, Type(821, 826, const Foo&) foo), Parameter(837, 845, Type(837, 840, Bar*) bar)]), 0x00, None in FooNS)
VariableDeclaration(848, 854, Type(848, 854, extern const ws::RMM*) full)
Function(876, 880, Type(876, 880, void) Function2([Parameter(891, 909, Type(891, 896, const Local&) local)]), 0x00, None in FooNS)
Class(922, 931, FooModule, [Type(941, 952, XSS::Module)], [Function(966, 975, None FooModule([]), 0x08, [])])
Class(1029, 1039, ProtoArray, None, None)
Class(1048, 1058, Experiment, None, [Function(1072, 1082, None Experiment([]), 0x08, None), Struct(1096, 1099, Foo, None, [Function(1106, 1110, Type(1106, 1110, bool) operator([Parameter(1123, 1135, Type(1123, 1128, const Foo*) a), Parameter(1137, 1149, Type(1137, 1142, const Foo*) b)]), 0x00, [Token('return', 1159, 1165), Token('true', 1166, 1170), Token(';', 1170, 1171)])])])
Enum(1226, 1227, nn, {[Token('bar', 1230, 1233), Token(',', 1233, 1234)]} in noname)
Include(1276, 1294, "bar.h")
Include(1301, 1320, <string>)
Class(1334, 1337, Abc::Xyz, None, [])
Class(1354, 1358, Fool, [Type(1368, 1371, Bar), Type(1380, 1383, Baz)], [])
Function(1435, 1439, Type(1435, 1439, void) lkjsdfkd([]), 0x00, None in <anonymous>)
Class(1462, 1464, TT, [Type(1474, 1480, TTbase<[Type(1481, 1484, Bar)]>)], [])
VariableDeclaration(1491, 1495, Type(1491, 1495, * void) ptof)
Typedef(1528, 1532, Setter, [None, Token('(', 1532, 1533), Token('*', 1533, 1534), Token('Setter', 1534, 1540), Token(')', 1540, 1541), Token('(', 1541, 1542), Token('int', 1542, 1545), Token('*', 1545, 1546), Token(',', 1546, 1547), Token('int', 1548, 1551), Token(')', 1551, 1552)])
VariableDeclaration(1572, 1576, Type(1572, 1576, Fool data[]) data = {NULL,})
Class(1604, 1611, def_var, None, [VariableDeclaration(1616, 1622, Type(1616, 1622, static const int) kConstant1 = 5), VariableDeclaration(1651, 1657, Type(1651, 1657, static const int) kConstant = (kConstant2-1)/4)])
Include(1706, 1727, "not-used.h")
Class(1747, 1763, LaunchNameEquals, None, [Function(1778, 1782, Type(1778, 1782, bool) operator([Parameter(1794, 1806, Type(1794, 1798, void*) launch)]), 0x00, [Token('return', 1814, 1820), Token('true', 1821, 1825), Token(';', 1825, 1826)] in <anonymous>)] in <anonymous>)
Enum(1857, 1858, None, {[Token('boo', 1858, 1861)]})
Class(1871, 1880, Blashdfdf, None, [Function(1895, 1899, Type(1895, 1899, bool) operator([Parameter(1911, 1923, Type(1911, 1915, void*) launch)]), 0x05, None)])
VariableDeclaration(1940, 1945, Type(1940, 1945, const char kBS[]) kBS = "~!@#$%^&*()-+{}|`[]\\:\";'<>?,./ =\t\r\n\v\f\177")
Class(2018, 2026, FooOnYou, None, [VariableDeclaration(2031, 2035, Type(2031, 2035, string & void) field_)])
Function(2094, 2098, Type(2094, 2098, void) AnnotatedFunction([]), 0x40, None)
Function(2146, 2150, Type(2146, 2150, void) RealAnnotatedFunction([]), 0x20, None)
Class(2208, 2217, TestAttrs, None, [Function(2222, 2226, Type(2222, 2226, void) AnnotatedMethod([]), 0x40, None), Function(2274, 2278, Type(2274, 2278, void) RealAnnotatedMethod([]), 0x20, None)])
Typedef(2346, 2355, IplImage, [Struct(2346, 2355, _IplImage, None, None)])
Struct(2386, 2388, AR, None, None in <anonymous>)
Class(2396, 2398, BS, None, [] in <anonymous>)
Class(2413, 2419, OpCode, None, [Enum(2447, 2448, Operator, {[Token('ID', 2453, 2455), Token('=', 2456, 2457), Token('0', 2458, 2459), Token(',', 2459, 2460), Token('CASE', 2465, 2469), Token('=', 2470, 2471), Token('1', 2472, 2473), Token(',', 2473, 2474), Token('DEL', 2479, 2482), Token('=', 2483, 2484), Token('2', 2485, 2486), Token(',', 2486, 2487), Token('INS', 2492, 2495), Token('=', 2496, 2497), Token('3', 2498, 2499), Token(',', 2499, 2500)]}), VariableDeclaration(2524, 2526, Type(2524, 2526, Operator) op)])
Class(2538, 2549, MemoryCache, None, [VariableDeclaration(2554, 2561, Type(2554, 2561, mutable Mutex) index_mutex_), VariableDeclaration(2591, 2601, Type(2591, 2601, struct CacheEntry*) index_)])
Struct(2623, 2627, lala, [Type(2630, 2643, dense_hashmap<[Type(2644, 2647, int)]>)], [])
Function(2759, 2763, Type(2759, 2763, void) Atmak([]), 0x80, None)
Enum(2802, 2803, Status, {None})
Enum(2821, 2822, RPC::Status2, {None})
Typedef(2838, 2841, XYZ123, [Class(2838, 2841, ABC, None, None)])
Typedef(2864, 2866, LP, [Class(2864, 2866, CB<Ev*,constAEv&>, None, None)])
Typedef(2902, 2906, JeeziePeezy, [Class(2902, 2906, OhMy, None, [])])
Struct(2959, 2972, HashIndexSpec, None, [Function(2977, 2990, None HashIndexSpec([]), 0x08, [])])
Class(3037, 3049, ThreadShared<RES>::Accessor, None, [])
Class(3086, 3092, FooStr, None, [VariableDeclaration(3097, 3101, Type(3097, 3101, char sizeof mystr(mystr)[]) ???)])
Class(3134, 3143, FooStruct, None, [VariableDeclaration(3155, 3162, Type(3155, 3162, struct in_addr) addr_)])
VariableDeclaration(3237, 3250, Type(3237, 3250, Enum(3179, 3180, None, {[Token('UNKNOWN', 3180, 3187), Token('=', 3188, 3189), Token('0', 3190, 3191), Token(',', 3191, 3192), Token('GARBAGE_COLLECTING', 3193, 3211), Token(',', 3211, 3212), Token('NOT_GARBAGE_COLLECTING', 3213, 3235)]})) garbage_state)
Class(3259, 3270, InlineClass, None, [VariableDeclaration(3281, 3285, Type(3281, 3285, class File*) file_), Function(3302, 3306, Type(3302, 3307, File*) Method1([]), 0x00, None), Function(3321, 3325, Type(3321, 3326, File*) Method2([]), 0x00, None)])
Typedef(3360, 3362, FT, [Enum(3360, 3362, A::D, {None})])
Include(3380, 3401, "dir//bar.h")
Function(3403, 3408, Type(3403, 3415, const string) single_backslash([Parameter(3433, 3437, Type(3433, 3437, "\\") ???)]), 0x00, None)
VariableDeclaration(3448, 3457, Type(3448, 3457, Struct(3448, 3457, BitFields, None, [VariableDeclaration(3464, 3467, Type(3464, 3467, int) done)])) flags_)
Class(3491, 3494, MyF, None, [Friend(3530, 3535, [Token('class', 3530, 3535), Token('shared_ptr', 3536, 3546)])])
Class(3558, 3563, Colon, None, [Function(3576, 3578, Type(3576, 3581, ::Foo) GetFoo([]), 0x02, None), Function(3596, 3599, Type(3596, 3599, Foo) GetFoo2([]), 0x00, None)])
VariableDeclaration(3615, 3621, Type(3615, 3621, string) foo = "\nX=\"/*\"")
VariableDeclaration(3650, 3658, Type(3650, 3658, struct streamer::Key) GlobalKey)
Typedef(3683, 3689, TimeFunc, Type(3683, 3711, time_t(TimeFunc)(time_t)*))
Typedef(3721, 3725, dummy_des_key_schedule, [None, Token('dummy_des_key_schedule', 3726, 3748), Token('[', 3748, 3749), Token('128', 3749, 3752), Token(']', 3752, 3753)])
Class(3762, 3766, Copy, None, [Function(3779, 3783, None Copy([]), 0x08, None), VariableDeclaration(3805, 3809, Type(3805, 3809, Copy&) operator = (constCopy&other))])
Struct(3866, 3870, Flag, None, None)
Class(3878, 3890, StructMethod, None, [Function(3895, 3900, Type(3895, 3908, struct Flag*) flag([]), 0x00, None)])
Class(3944, 3960, TemplateCtorDtor, None, [Function(3974, 3990, None TemplateCtorDtor([Parameter(3994, 4000, Type(3994, 3995, T*) ptr)]), 0x08, None), Function(4006, 4022, None TemplateCtorDtor([]), 0x10, None)])
VariableDeclaration(4058, 4062, Type(4058, 4062, Type Singleton<[Type(4074, 4078, Type)]>*) instance_ = NULL)
Function(4100, 4103, Type(4100, 4103, int) CreateFakeResults([Parameter(4122, 4130, Type(4122, 4126, bool) add), Parameter(4132, 4143, Type(4132, 4137, float) score = 0.5)]), 0x00, None)
Function(4153, 4156, Type(4153, 4177, map<[Type(4157, 4163, string), Type(4165, 4175, const Flag*)]>) ReturnFlags([]), 0x00, None)
VariableDeclaration(4194, 4200, Type(4194, 4200, vector<[Type(4201, 4207, string*)]>) string_list)
Class(4230, 4233, Foo::Bar, None, [Function(4241, 4244, None Bar([]), 0x08, [Token('XXX', 4249, 4252), Token('(', 4252, 4253), Token('1', 4253, 4254), Token(')', 4254, 4255), Token('<<', 4256, 4258), Token('"should work"', 4259, 4272), Token(';', 4272, 4273)])])
